### PR TITLE
Archive 🗂

### PR DESCRIPTION
**Pull request by pierreericgarcia**
Face à l'abondance d'emojis symbolisant un classeur, une intercalaire ou un rangement, j'hésite :

1. 🗃
2. 🗄
3. 📁
4. 📂
5. 🗂
6. 📒

### CODE REVIEW

**Comment by pierreericgarcia** [comment]
[ETQ U, je peux archiver une habilitation](https://trello.com/c/mxH9KEqX/1690-etq-u-je-peux-archiver-une-habilitation)

**Comment by Isalafont** [comment]
[ETQ Demandeur ayant au moins une demande en A modifier depuis 6 mois ou plus, je reçois un email me prévenant de sa suppression prochaine](https://trello.com/c/LQjLpPg3/1707-etq-demandeur-ayant-au-moins-une-demande-en-a-modifier-depuis-6-mois-ou-plus-je-re%C3%A7ois-un-email-me-pr%C3%A9venant-de-sa-suppression-p)

File: backend/spec/controllers/enrollments/export_spec.rb [hunk 1]
```diff
@@ -60,7 +60,7 @@
     end
 
     it "does not make database queries" do
-      expect { subject }.to make_database_queries(count: 3)
+      expect { subject }.to make_database_queries(count: 4)
```
Comment (rdubigny) [inline backend/spec/controllers/enrollments/export_spec.rb:63]: This is a performance regression. We must keep the number of database query to 3.

File: backend/app/policies/enrollment_policy.rb [hunk 2]
```diff
@@ -15,17 +19,17 @@ def create?
   end
 
   def update?
+    if record.status_archived?
```
Comment (rdubigny) [inline backend/app/policies/enrollment_policy.rb:22]: this condition is not necessary as the condition `(record.status_draft? || record.status_changes_requested?)` below imply it

File: backend/app/policies/enrollment_policy.rb [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 class EnrollmentPolicy < ApplicationPolicy
   def show?
+    if record.status_archived?
```
Comment (rdubigny) [inline backend/app/policies/enrollment_policy.rb:5]: we could also keep the onliner synthax:

`!record.status_archived? && (user.is_member?(record) || user.is_reporter?(record))`

**Review by rdubigny — CHANGES_REQUESTED** [review]

File: backend/app/views/enrollment_mailer/archive.text.erb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+Bonjour,
+
+Votre habilitation a été archivé.
+
+Pour consulter cette habilitation, veuillez cliquer sur le lien suivant <%= @url %> .
```
[the remark this answers is not in the document]
Comment (pierreericgarcia) [reply backend/app/views/enrollment_mailer/archive.text.erb]: This mail was deleted.

File: frontend/src/components/templates/Form/SubmissionPanel/index.tsx [hunk 5]
```diff
@@ -81,6 +81,17 @@ const SubmissionPanel: FunctionComponent<Props> = ({
             Voulez vous continuer ?
           </ConfirmationModal>
         )}
+      {pendingEvent &&
+        eventConfigurations[pendingEvent].prompt ===
+          PromptType.confirm_archive && (
+          <ConfirmationModal
+            title="L'archivage d'une habilitation est irréversible"
```
Comment (rdubigny) [inline frontend/src/components/templates/Form/SubmissionPanel/index.tsx:88]: Somehow, isn't this a lie ? 🤔

File: backend/app/policies/enrollment_policy.rb [hunk 6]
```diff
@@ -72,6 +68,10 @@ def request_changes?
     record.can_request_changes_status? && user.is_instructor?(record.target_api)
   end
 
+  def archive?
+    (record.can_archive_status? && user.is_instructor?(record.target_api)) || user.is_administrator?
```
Comment (rdubigny) [inline backend/app/policies/enrollment_policy.rb:72]: record.can_archive_status? should always be needed, even from administrator.

**Review by rdubigny — CHANGES_REQUESTED** [review]

**Review by Isalafont — APPROVED** [review]

**Comment by Isalafont** [comment]
Great PR :tada: !! 
For me it's good and ready to be merged on master ! 🚀

**Review by Isalafont — APPROVED** [review]

**Pull request merged**